File: 0101_0200/0118/0118.py
def z(n, k):
    return 1 if n == 1 else 1 + (z(n - 1, k) + k - 1) % n

with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, k = (int(x) for x in f.readline().split())
    q.write(str(z(n, k)))

File: 0101_0200/0118/0118.c
#include <stdio.h>

int z(int n, int m)
{
	return n == 1 ? 1 : 1 + (z(n - 1, m) + m - 1) % n;
}

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int k, n;

	fscanf(f, "%d%d", &n, &k);
	fprintf(q, "%d", z(n, k));
	return 0;
}
